Support JDK25

taeyeon-Kim opened the pull request:
resolved: #163

sohyun-ku reviewed (commented):
It looks like the formatting diff has grown significantly due to changes in the ktlint rule configuration.
Since this configuration affects the overall code style, it would be helpful for review if the logic changes and formatting changes could be separated into different commits.

kojandy commented on scavenger-agent-java/build.gradle.kts:
```diff
@@ -67,18 +68,19 @@ repositories {
+    implementation("org.ow2.asm:asm-tree:9.9")
     implementation("com.squareup.okhttp3:okhttp:3.14.9")
     implementation("com.google.protobuf:protobuf-java-util:${property("protobufVersion")}")
     implementation("io.grpc:grpc-stub:${property("grpcVersion")}")
```
```suggestion
```

As `io.grpc:grpc-stub` is declared as a transitive dependency in [scavenger-model](https://github.com/naver/scavenger/pull/164/changes#diff-d75831eab0844228af39655dfdb533f925ed7d64bfe252c5ab9a4c74e59c9a2eR17), it seems this line may no longer be necessary, similar to [the changes](https://github.com/naver/scavenger/pull/164/changes#diff-c1b148fe1efaf0912fa6bed1fb9632e2c2c8d5f1645338e131dfd01f09b5630bL46) in scavenger-collector.

taeyeon-Kim replied:
Thanks, applied in https://github.com/naver/scavenger/commit/e531ed53af335c64962284b9daab795741a967d1

kojandy commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,5 +1,7 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.9-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-9.1.0-bin.zip
```
https://github.com/naver/scavenger/blob/b78a45580fdae90fd1bc7daa8ab806957f87dedc/settings.gradle.kts#L17-L19

[org.gradle.toolchains.foojay-resolver-convention](https://plugins.gradle.org/plugin/org.gradle.toolchains.foojay-resolver-convention) needs to be upgraded to version 1.0.0 for gradle 9 compatibility.

taeyeon-Kim replied:
Thanks, applied in https://github.com/naver/scavenger/commit/e531ed53af335c64962284b9daab795741a967d1

kojandy commented on scavenger-agent-java/build.gradle.kts:
```diff
@@ -67,18 +68,19 @@ repositories {
+    testImplementation(platform("org.mockito:mockito-bom:5.17.0"))
     testImplementation("org.junit.jupiter:junit-jupiter")
-    testImplementation("org.assertj:assertj-core:3.22.0")
+    testRuntimeOnly("org.junit.platform:junit-platform-launcher")
```
https://github.com/naver/scavenger/blob/b78a45580fdae90fd1bc7daa8ab806957f87dedc/scavenger-old-model/build.gradle.kts#L8-L16

`junit-platform-launcher` should also be added to `scavenger-old-model/build.gradle.kts` as well.

taeyeon-Kim replied:
Thanks, applied in https://github.com/naver/scavenger/commit/e531ed53af335c64962284b9daab795741a967d1

kojandy commented on scavenger-api/build.gradle.kts:
```diff
@@ -3,24 +3,24 @@ import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
     id("org.jetbrains.kotlin.plugin.spring") version kotlinVersion
     id("org.springframework.boot") version springBootVersion
-    id("com.adarshr.test-logger") version "3.0.0"
+    id("com.adarshr.test-logger") version "4.0.0"
```
https://github.com/naver/scavenger/blob/b78a45580fdae90fd1bc7daa8ab806957f87dedc/scavenger-old-model/build.gradle.kts#L4

This change should also be applied to `scavenger-old-model/build.gradle.kts`.

taeyeon-Kim replied:
Thanks, applied in e531ed5

taeyeon-Kim committed:
Upgrade dependencies for JDK 25 and Spring Boot 3.5.9

- Gradle: 9.1.0
- Spring Boot: 3.5.9
- Kotlin: 2.3.0
- Spring Cloud: 2025.0.1
- Armeria: 1.31.3
- ByteBuddy: 1.17.8
- gRPC: 1.68.0
- JUnit: 5.12.2
- Mockito: 5.17.0
- AssertJ: 3.27.6

taeyeon-Kim committed:
Address PR review comments

- Remove redundant grpc-stub dependency (transitive from scavenger-model)
- Upgrade foojay-resolver-convention to 1.0.0 for Gradle 9 compatibility
- Add junit-platform-launcher to scavenger-old-model
- Upgrade test-logger plugin to 4.0.0 in scavenger-old-model

taeyeon-Kim committed:
Fix compilation errors for Kotlin 2.3.0 and updated dependencies

- Update DelegatableJdbcRepository return types (MutableIterable -> MutableList)
- Add writeIsolationLevel parameter to SqlRayQueryLogEntryCreator
- Fix BridgeService property initialization for Kotlin 2.3.0

taeyeon-Kim commented:
> It looks like the formatting diff has grown significantly due to changes in the ktlint rule configuration. Since this configuration affects the overall code style, it would be helpful for review if the logic changes and formatting changes could be separated into different commits.

@sohyun-ku  Added ktlint rule exclusions in ca97bee to keep the existing code formatting
